fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/AccountNumberGenerator.java

```diff
@@ -165,7 +153,7 @@ private String generateAccountNumber(Map<String, String> propertyMap, AccountNum
                 prefix = prefix.substring(0, Math.min(prefix.length(), 10));
             }
             if (accountNumberPrefixType.getValue().equals(AccountNumberPrefixType.PREFIX_SHORT_NAME.getValue())) {
-                Integer prefixLength = prefix.length();
+                Integer prefixLength = prefix == null ? 10 : prefix.length();
```

Comment:
Is this needed for the externalId work? Either revert this or cover with tests.

Reply:
This is a Potential null pointer access, but If that makes you happy, It has been reverted